rllib/examples/inference/policy_inference_after_training_w_connector.py

```diff
@@ -0,0 +1,265 @@
+
+
+parser = add_rllib_example_script_args(default_reward=200.0)
+parser.set_defaults(
+    # Make sure that - by default - we produce checkpoints during training.
+    checkpoint_freq=1,
+    checkpoint_at_end=True,
+    # Use StatelessCartPole by default.
+    env="stateless-cart",
+)
+parser.add_argument(
+    "--explore-during-inference",
+    action="store_true",
+    help="Whether the trained policy should use exploration during action "
+    "inference.",
+)
+parser.add_argument(
+    "--num-episodes-during-inference",
+    type=int,
+    default=10,
+    help="Number of episodes to do inference over (after restoring from a checkpoint).",
+)
+
+
+if __name__ == "__main__":
+    args = parser.parse_args()
+
+    assert (
+        args.enable_new_api_stack
+    ), "Must set --enable-new-api-stack when running this script!"
+
+    base_config = (
+        get_trainable_cls(args.algo)
+        .get_default_config()
+        .training(
+            num_sgd_iter=6,
+            lr=0.0003,
+            vf_loss_coeff=0.01,
+        )
+        # Add an LSTM setup to the default RLModule used.
+        .rl_module(model_config_dict={"use_lstm": True})
+    )
+
+    print("Training LSTM-policy until desired reward/timesteps/iterations. ...")
+    results = run_rllib_example_script_experiment(base_config, args)
+
+    print("Training completed. Creating an env-loop for inference ...")
+
+    print("Env ...")
+    env = _env_creator(base_config.env_config)
+
+    # We build the env-to-module pipeline here manually, but feel also free to build it
+    # through the even easier:
+    # `env_to_module = base_config.build_env_to_module_connector(env=env)`, which will
+    # automatically add all default pieces necessary (for example the
+    # `AddStatesFromEpisodesToBatch` component b/c we are using a stateful RLModule
+    # here).
+    print("Env-to-module ConnectorV2 ...")
+    env_to_module = EnvToModulePipeline(
+        input_observation_space=env.observation_space,
+        input_action_space=env.action_space,
+        connectors=[
+            AddObservationsFromEpisodesToBatch(),
+            AddStatesFromEpisodesToBatch(),
+            BatchIndividualItems(),
+            NumpyToTensor(),
+        ],
+    )
+
+    # Create the RLModule.
+    # Get the last checkpoint from the above training run.
+    best_result = results.get_best_result(
+        metric=f"{ENV_RUNNER_RESULTS}/{EPISODE_RETURN_MEAN}", mode="max"
+    )
+    # Create new Algorithm and restore its state from the last checkpoint.
+    rl_module = RLModule.from_checkpoint(
+        os.path.join(
+            best_result.checkpoint.path,
+            "learner",
+            "module_state",
+            DEFAULT_MODULE_ID,
+        )
+    )
+    print("RLModule restored ...")
+
+    # For the module-to-env pipeline, we will use the convenient config utility.
+    print("Module-to-env ConnectorV2 ...")
+    module_to_env = base_config.build_module_to_env_connector(env=env)
```

Comment:
Why can we build this easily from the alto and the env-to-module not?

Reply:
We can do both with `config.build_[env_to_module/module_to_env]_connector`, just wanted to show both ways.